roles/gateway_preupgrade_analyzer/README.md

```diff
@@ -1,23 +1,64 @@
-Pre-Upgrade-Upgrader
+Preupgrade Analyzer
 ======================
 
-This role will inform which customizations on the Gateway will be migrated automatically, and which ones customer will need to migrate themself.
+This role will generate pre-upgrade analyzer report for all servers listed in hosts file under [gateway] section. 
+Please review the pre-upgrade analyzer report before upgrading the gateway.
 
 Requirements
 ------------
-Target Gateway is started.
-Provide Gateway Database admin user and password, gateway user and password.
-Have full access both Gateway (ssg) database. 
-MySql Shell Checker is installed on the control node
+* ssh connect to remote source gateway as user ssgconfig using encrypted password. 
+    Specify source gateway's hostname/ip in the inventories/<staging/production>/hosts file [gateway] section.
+    ei. inventories/sample/hosts
+    
+* In order to run MySQL 8 compatibility checker, login to source gateway manually (defined in hosts file [gateway_primary_db] and [gateway_failover_db] section) as database admin user and run the following command
+    ```
+    GRANT ALL PRIVILEGES ON *.* TO 'gateway'@'%'; flush privileges;
+    ```
+  Make sure to revoke the privileges after the tasks is completed
+    ```
+    REVOKE ALL PRIVILEGES ON *.* from 'gateway'@'%'; flush privileges;
+    ```
+
 
 Role Variables
 --------------
-access both Gateway (ssg) and OTK database using mysql user gateway. 
+Under inventories/xxx/group_vars/all/vars
+* database_user - Database Username default as gateway
+* database_pass -  Database Password
+* database_name - Gateway Database Name
+* database_port - Database Port default as 3306
+
+Under roles/gateway_preupgrade_analyzer/vars/main.yml
+* remote_report_temp_dir - remote directory to store report files.
+* gateway_modular_assertions_dir - target gateway modular assertions directory use '/opt/SecureSpan/Gateway/runtime/modules/assertions'
```

Comment:
comment only: I believe this folder will never change and should not have been made configurable.